### DISCUSSION TITLE
[Question] charge_state Battery range

### DISCUSSION THREAD

**Discussion by pinloop** [timdorr/tesla-api#412] [Q&A]
Not sure if this is the correct place, but in the `/api/1/vehicles/{id}/data_request/charge_state`, what is the difference between these three responses:

"battery_range": 224.47
"est_battery_range": 171.24
"ideal_battery_range": 280.59

**Comment by pinloop** [comment]
Bumping this.. Anyone?

**Reply by nikemeal** [reply]
i *think* its the similar/same as the usage chart in the car.  the chart that shows you range over last X miles etc.  so i'd guess ideal range is best case scenario, battery range is what what it thinks you will get and estimated range is based on how you've been driving recently.
as i said, just my thought on it.  might be best to get the values from the api then look at the screen in car to see if any match?
